### ISSUE TITLE
handle diacritics gracefully

### ISSUE THREAD

**Issue by arrocke** [arrocke/gloss-translation#84]
## acceptance criteria
* glosses with the same diacritics are treated as the same word
* autocomplete matches even when diacritics are in a different order

**Comment by Pertempto** [comment]
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize

**Comment by Pertempto** [comment]
> glosses with the same diacritics are treated as the same word


What do you mean by this @arrocke? I'm not getting it.

**Comment by Pertempto** [comment]
Maybe I should wait to do this one until after #106

**Comment by arrocke** [comment]
> glosses with the same diacritics are treated as the same word

A letter can have multiple diacritics (ie in Hebrew שָּׁ is formed from the character \u05e9 plus these three diacritics in any order \u05c1, \u05bc, \u05b8). This means there are 6 ways of encoding that letter with those diacritics, but they render the same to the user. When a user types in a gloss, we need to connect to an existing matching gloss even if the diacritics are in a different order. Otherwise the number of ways a word has been glossed will be artificially inflated with what appears to be duplicates.

**Comment by Pertempto** [comment]
Ok great! It appears that the built-in string normalization function handles that well. It'll just be a matter of figuring out how to modify the auto complete to support this. I'm going to put this back to "Ready" until I complete #106.

**Comment by Pertempto** [comment]
Handled in #119

**Comment by arrocke** [comment]
I think there may be a few additional things to this that can either be addressed in #119 or another:
1. We'll want to normalize on the backend as well under the assumption of not assuming the client has properly processed the input
2. We'll want to use 'NFD' normalization since that moves the diacritics to separate bytes in the string. That allows for searches that ignore the diacritics.
3. In the autocomplete, we want to match strings on just the characters, ignoring diacritics. For example if you were working with Hebrew, נֹתֵן (giving), and נָתַו (he gives) should both be matched by נתן without any of the vowel points.

**Comment by Pertempto** [comment]
Ok I'll do it in a separate pr
